Improve device pixel accuracy for Touch Events

xingri opened the pull request:
#### 628bfae1ad5915f3ef863e76dc0ef5082925e13f
<pre>
Improve device pixel accuracy for Touch Events
<a href="https://bugs.webkit.org/show_bug.cgi?id=133180">https://bugs.webkit.org/show_bug.cgi?id=133180</a>

Reviewed by NOBODY (OOPS!).

Previously TouchEvents had their co-ordinates truncated to integers.
By supporting sup-pixel precision we get smoother dragging and line
drawing at slow speeds on high-dpi devices or when zoomed.

This is a cherry-pick from Blink:
- <a href="https://codereview.chromium.org/298133003">https://codereview.chromium.org/298133003</a>

* LayoutTests/fast/events/touch/resources/frame-touchevent-forwarder.html: Added.
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll-expected.txt:
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll.html:
* LayoutTests/fast/events/touch/touch-fractional-coordinates-expected.txt: Added.
* LayoutTests/fast/events/touch/touch-fractional-coordinates.html: Added.
* Source/WebCore/dom/Document+Touch.idl:
* Source/WebCore/dom/DocumentTouch.cpp:
(WebCore::DocumentTouch::createTouch):
* Source/WebCore/dom/DocumentTouch.h:
* Source/WebCore/dom/Touch.cpp:
(WebCore::contentsOffset):
(WebCore::scaledLocation):
(WebCore::Touch::Touch):
(WebCore::Touch::cloneWithNewTarget const):
(WebCore::contentsX): Deleted.
(WebCore::contentsY): Deleted.
* Source/WebCore/dom/Touch.h:
(WebCore::Touch::create):
(WebCore::Touch::clientX const):
(WebCore::Touch::clientY const):
(WebCore::Touch::screenX const):
(WebCore::Touch::screenY const):
(WebCore::Touch::pageX const):
(WebCore::Touch::pageY const):
(WebCore::Touch::webkitRadiusX const):
(WebCore::Touch::webkitRadiusY const):
* Source/WebCore/dom/Touch.idl:
* Source/WebCore/dom/wpe/PointerEventWPE.cpp:
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::SyntheticTouchPoint::SyntheticTouchPoint):
(WebCore::EventHandler::handleTouchEvent):
* Source/WebCore/platform/PlatformTouchPoint.h:
(WebCore::PlatformTouchPoint::PlatformTouchPoint):
(WebCore::PlatformTouchPoint::screenPos const):
(WebCore::PlatformTouchPoint::pos const):
(WebCore::PlatformTouchPoint::radius const):
(WebCore::PlatformTouchPoint::radiusX const): Deleted.
(WebCore::PlatformTouchPoint::radiusY const): Deleted.
* Source/WebCore/platform/ScrollView.cpp:
(WebCore::ScrollView::viewToContents const):
(WebCore::ScrollView::windowToContents const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::convertFromContainingWindow const):
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::FloatPoint::scaledBy const):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::FloatSize::scaledBy const):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformTouchPoint::WebKit2PlatformTouchPoint):
* Source/WebKit/Shared/WebTouchEvent.h:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::location const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateTouchEventTracking):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/628bfae1ad5915f3ef863e76dc0ef5082925e13f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65246 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44615 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17860 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69270 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15852 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/67364 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52397 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16134 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52416 "Found 2 new test failures: fast/events/touch/touch-coords-in-zoom-and-scroll.html fast/events/touch/touch-fractional-coordinates.html (failure)") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10976 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68312 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41232 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56474 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33039 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37903 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13845 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-masking/clip-path/animations/clip-path-animation-font-size-mixed-change.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14729 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59785 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14185 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70975 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9198 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13663 "Found 1 new test failure: imported/w3c/web-platform-tests/media-source/mediasource-addsourcebuffer-mode.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59741 "Found 1 new test failure: fast/events/touch/touch-fractional-coordinates.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9230 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56535 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60015 "Found 2 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/children-changed, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/basic-hierarchy (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7601 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1272 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40425 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/41502 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42683 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41246 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

lilyspiniolas commented on Source/WebCore/dom/DocumentTouch.cpp:
```diff
@@ -37,19 +37,38 @@
+        rotationAngle = 0;
+    if (!std::isfinite(force))
+        force = 0;
+
```
Nit: Using ternary operators here instead of if-statements might be appealing, i.e `pageX = std::isfinite(pageX) ? pageX : 0`

xingri replied:
@lilyspiniolas  Thanks for the review. Just updated the PR.

xingri committed:
Improve device pixel accuracy for Touch Events
https://bugs.webkit.org/show_bug.cgi?id=133180

Reviewed by NOBODY (OOPS!).

Previously TouchEvents had their co-ordinates truncated to integers.
By supporting sup-pixel precision we get smoother dragging and line
drawing at slow speeds on high-dpi devices or when zoomed.

This is a cherry-pick from Blink:
- https://codereview.chromium.org/298133003

* LayoutTests/fast/events/touch/resources/frame-touchevent-forwarder.html: Added.
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll-expected.txt:
* LayoutTests/fast/events/touch/touch-coords-in-zoom-and-scroll.html:
* LayoutTests/fast/events/touch/touch-fractional-coordinates-expected.txt: Added.
* LayoutTests/fast/events/touch/touch-fractional-coordinates.html: Added.
* Source/WebCore/dom/Document+Touch.idl:
* Source/WebCore/dom/DocumentTouch.cpp:
(WebCore::DocumentTouch::createTouch):
* Source/WebCore/dom/DocumentTouch.h:
* Source/WebCore/dom/Touch.cpp:
(WebCore::contentsOffset):
(WebCore::scaledLocation):
(WebCore::Touch::Touch):
(WebCore::Touch::cloneWithNewTarget const):
(WebCore::contentsX): Deleted.
(WebCore::contentsY): Deleted.
* Source/WebCore/dom/Touch.h:
(WebCore::Touch::create):
(WebCore::Touch::clientX const):
(WebCore::Touch::clientY const):
(WebCore::Touch::screenX const):
(WebCore::Touch::screenY const):
(WebCore::Touch::pageX const):
(WebCore::Touch::pageY const):
(WebCore::Touch::webkitRadiusX const):
(WebCore::Touch::webkitRadiusY const):
* Source/WebCore/dom/Touch.idl:
* Source/WebCore/dom/wpe/PointerEventWPE.cpp:
(WebCore::PointerEvent::PointerEvent):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::SyntheticTouchPoint::SyntheticTouchPoint):
(WebCore::EventHandler::handleTouchEvent):
* Source/WebCore/platform/PlatformTouchPoint.h:
(WebCore::PlatformTouchPoint::PlatformTouchPoint):
(WebCore::PlatformTouchPoint::screenPos const):
(WebCore::PlatformTouchPoint::pos const):
(WebCore::PlatformTouchPoint::radius const):
(WebCore::PlatformTouchPoint::radiusX const): Deleted.
(WebCore::PlatformTouchPoint::radiusY const): Deleted.
* Source/WebCore/platform/ScrollView.cpp:
(WebCore::ScrollView::viewToContents const):
(WebCore::ScrollView::windowToContents const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::convertFromContainingWindow const):
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/graphics/FloatPoint.h:
(WebCore::FloatPoint::scaledBy const):
* Source/WebCore/platform/graphics/FloatSize.h:
(WebCore::FloatSize::scaledBy const):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformTouchPoint::WebKit2PlatformTouchPoint):
* Source/WebKit/Shared/WebTouchEvent.h:
(WebKit::WebPlatformTouchPoint::WebPlatformTouchPoint):
(WebKit::WebPlatformTouchPoint::location const):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateTouchEventTracking):
